service/frontend/workflow_handler.go

```diff
@@ -2423,6 +2423,10 @@ func (wh *WorkflowHandler) DescribeTaskQueue(ctx context.Context, request *workf
 		request.TaskQueueType = enumspb.TASK_QUEUE_TYPE_WORKFLOW
 	}
 
+	if len(request.TaskQueueTypes) == 0 {
+		request.TaskQueueTypes = []enumspb.TaskQueueType{enumspb.TASK_QUEUE_TYPE_WORKFLOW, enumspb.TASK_QUEUE_TYPE_ACTIVITY}
+	}
+
```

Comment:
should we add a validation here to make sure if enhanced mode is enabled the partition is not sticky or non-root?

Reply:
I was thinking to do the validation of making sure the node is root within matching_engine.go , because that is where we are able to call isRoot() on a partition. If we wanted to check whether the task queue is root at the frontend layer, we would have to check the task queue name there, and we don't have access to the taskQueuePartitionPrefix and taskQueueSeparatorCharacter (or whatever they're called) at the frontend level.

My thought is to validate anything we are able to at the frontend level, but if we need to pull in additional info to validate something, then that validation can happen inside the matching engine

Reply:
agreed we should do it with PartitionFromProto